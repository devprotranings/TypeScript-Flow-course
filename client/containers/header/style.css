.header {
  height: 50px;
  padding: 10px 20%;
  background-color:#fff;
  margin-bottom: 10px;
  box-shadow: 0 4px 2px -2px gray;
}