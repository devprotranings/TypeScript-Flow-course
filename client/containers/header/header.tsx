import * as React from 'react';
import './style.css';

import {debounce} from '../../helpers';

export interface IProps {
  search(term: string): void;
}

export class Header extends React.Component<IProps> {

  @debounce(500)
  public onSearch(term: string) {
    this.props.search(term);
  }

  public render(): JSX.Element {
    return <div className='row header'>
      Search: <input type='text' onChange={(e) => this.onSearch.call(this, e.target.value)}/>
    </div>;
  }
}
