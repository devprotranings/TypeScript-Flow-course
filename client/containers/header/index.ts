import { connect } from 'react-redux';
import { Dispatch } from 'redux';
import { searchByTerm } from '../../store/actions/search-term.action';
import { Header } from './header';

const mapDispatchToProps = (dispatch: Dispatch) => ({
  search: (term: string) => dispatch( searchByTerm(term)),
});

export default connect(null, mapDispatchToProps)(Header as any);
