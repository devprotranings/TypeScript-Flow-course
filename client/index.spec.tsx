import { configure, mount, ReactWrapper, shallow, ShallowWrapper } from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import * as React from 'react';
import { Header,IProps } from './containers/header/header';
import { App } from './index';

configure({ adapter: new Adapter() });



describe('<App />', () => {
  it('shallow <App />', () => {
    const wrapper: ShallowWrapper = shallow(<App/>);
    expect(wrapper.find('div').length)
      .toBe(1);
  });

  it('shallow <Header />', () => {
    const mockSearch = jest.fn();
    const wrapper: ReactWrapper<IProps> = mount<IProps>(<Header search={mockSearch}/>);
    wrapper.find('input')
     // .simulate('—Åhange');
    expect(mockSearch.mock.calls.length).toBe(1);
  });
});
