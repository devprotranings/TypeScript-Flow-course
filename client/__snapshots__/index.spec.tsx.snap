// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> shallow <Header /> 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    search={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Search: ",
        <input
          onChange={[Function]}
          type="text"
        />,
      ],
      "className": "row header",
    },
    "ref": null,
    "rendered": Array [
      "Search: ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Search: ",
          <input
            onChange={[Function]}
            type="text"
          />,
        ],
        "className": "row header",
      },
      "ref": null,
      "rendered": Array [
        "Search: ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
